esphome/components/sml/__init__.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+import voluptuous as vol
+
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import uart
+from esphome.const import CONF_ID
+
+CODEOWNERS = ["@alengwenus"]
+
+DEPENDENCIES = ["uart"]
+
+sml_ns = cg.esphome_ns.namespace("sml")
+Sml = sml_ns.class_("Sml", cg.Component, uart.UARTDevice)
+
+CONF_SML_ID = "sml_id"
+CONF_OBIS = "obis"
```

Comment:
```suggestion
CONF_OBIS_CODE = "obis_code"
```

Nit: From what I can gather "obis" (object identification system) is the name of the _system_; the individual codes should then be called `obis_id`

Reply:
Changed it.